test/SleepRepository-test.js

```diff
@@ -0,0 +1,103 @@
+const { expect } = require('chai');
+const chai = require('chai');
+const expecet = chai.expect;
+// these are commented out because this class hasn't been created and will throw errors until then. Check line 68 also before testing.
+// Remove before committing after class has been created
+// const SleepRepository = require('../src/SleepRepository.js');
+
+describe('Sleep Repository', function() {
+  let sleepRepo;
+  let sleepData;
+
+  beforeEach(function() {
+    sleepData = [
+      {
+        "userID": 1,
+        "date": "2019/06/15",
+        "hoursSlept": 6.1,
+        "sleepQuality": 2.2
+      }, 
+      {
+        "userID": 1,
+        "date": "2019/06/16",
+        "hoursSlept": 4.1,
+        "sleepQuality": 3.8
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/17",
+        "hoursSlept": 8,
+        "sleepQuality": 2.6
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/18",
+        "hoursSlept": 10.4,
+        "sleepQuality": 3.1
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/19",
+        "hoursSlept": 10.7,
+        "sleepQuality": 1.2
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/20",
+        "hoursSlept": 9.3,
+        "sleepQuality": 1.2
+      },
+      {
+        "userID": 1,
+        "date": "2019/06/21",
+        "hoursSlept": 7.8,
+        "sleepQuality": 4.2
+      },
+      {
+        "userID": 2,
+        "date": "2019/06/20",
+        "hoursSlept": 7.0,
+        "sleepQuality": 3.9
+      },
+      {
+        "userID": 2,
+        "date": "2019/06/21",
+        "hoursSlept": 6.5,
+        "sleepQuality": 4.0
+      }
+    ];
+    // sleepRepo = new SleepRepository(sleepData);
+  });
+
+  it.skip('should be a function', function() {
+    expect(SleepRepository).to.be.be.a('function');
+  });
+
+  it.skip('should instantiate the Sleep Repository', function() {
+    expect(sleepRepo).to.be.an.instanceOf(SleepRepository);
+  });
+
+  it.skip('should hold all of the sleep objects', function() {
```

Comment:
Ok, so its a straightforward copy of the raw data!!! If the class methods work out easily, I should probably revise the `ActivityRepository` tests, which assume some fancier footwork when processing the data in the constructor